RecoTracker/TkSeedingLayers/src/HitExtractorSTRP.cc

```diff
@@ -116,49 +118,76 @@ void HitExtractorSTRP::cleanedOfClusters(const TkTransientTrackingRecHitBuilder&
                                          HitExtractor::Hits& hits,
                                          bool matched,
                                          unsigned int cleanFrom) const {
-  LogDebug("HitExtractorPIX") << "getting: " << hits.size() << " in input.";
-  edm::Handle<SkipClustersCollection> stripClusterMask;
-  if (maskCluster)
-    ev.getByToken(theSkipClusters, stripClusterMask);
   unsigned int skipped = 0;
   unsigned int projected = 0;
-  for (unsigned int iH = cleanFrom; iH < hits.size(); ++iH) {
-    assert(hits[iH]->isValid());
-    auto id = hits[iH]->geographicalId();
-    if (matched) {
-      bool replace;
-      ProjectedSiStripRecHit2D* replaceMe;
-      std::tie(replace, replaceMe) = skipThis(ttrhBuilder, *hits[iH], stripClusterMask);
-      if (replace) {
-        if (!replaceMe) {
-          LogDebug("HitExtractorSTRP") << "skipping a matched hit on :" << hits[iH]->geographicalId().rawId();
-          skipped++;
-        } else
-          projected++;
-        hits[iH].reset(replaceMe);
-        if (replaceMe == nullptr)
-          assert(hits[iH].empty());
-        else
-          assert(hits[iH].isOwn());
+  if (hasMatchedHits || hasRPhiHits || hasStereoHits) {
+    LogTrace("HitExtractorSTRP") << "getting " << hits.size() << " strip hit in input.";
+    edm::Handle<SkipClustersCollection> stripClusterMask;
+    if (maskCluster)
+      ev.getByToken(theSkipClusters, stripClusterMask);
+    for (unsigned int iH = cleanFrom; iH < hits.size(); ++iH) {
+      assert(hits[iH]->isValid());
+      auto id = hits[iH]->geographicalId();
+      if (matched) {
+        bool replace;
+        ProjectedSiStripRecHit2D* replaceMe;
+        std::tie(replace, replaceMe) = skipThis(ttrhBuilder, *hits[iH], stripClusterMask);
```

Comment:
this could use structured bindings to reduce verbosity

Reply:
Done.